runelite-client/src/main/java/net/runelite/client/plugins/hunter/HunterPlugin.java

```diff
@@ -296,6 +318,32 @@ public void onGameObjectSpawned(GameObjectSpawned event)
 		}
 	}
 
+	@Subscribe
+	public void onGameObjectChanged(GameObjectChanged event)
+	{
+		onGameObject(event.getPrevious(), event.getGameObject());
+	}
+
+	@Subscribe
+	public void onGameObjectDespawned(GameObjectDespawned event)
+	{
+		onGameObject(event.getGameObject(), null);
```

Comment:
why not just remove the object from the collection here? Rather than invoking the method with a null. 

Alternatively overload `onGameObject`.

You're having to add null guards to `onGameObject` just because you want to call it with a null.